data/spells/scripts/support/convince_creature.lua

```diff
@@ -0,0 +1,31 @@
+function onCastSpell(creature, variant, isHotkey)
+	local target = Creature(variant:getNumber())
+	if not target or not target:isMonster() then
+		creature:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
+		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local monsterType = target:getType()
+	if not getPlayerFlagValue(creature, PlayerFlag_CanConvinceAll) then
```

Comment:
You should probably get rid of such ``getPlayerFlagValue`` statements, we should not care about these flags particularities.